src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/sirius/SingleRowIdentificationTask.java

```diff
@@ -0,0 +1,218 @@
+import io.github.msdk.id.sirius.SiriusIdentificationMethod;
+import io.github.msdk.id.sirius.SiriusIonAnnotation;
+import io.github.msdk.util.IonTypeUtil;
+import java.text.NumberFormat;
+import java.util.LinkedList;
+import java.util.List;
+import net.sf.mzmine.datamodel.DataPoint;
+import net.sf.mzmine.datamodel.Feature;
+import net.sf.mzmine.datamodel.IonizationType;
+import net.sf.mzmine.datamodel.PeakListRow;
+import net.sf.mzmine.datamodel.RawDataFile;
+import net.sf.mzmine.datamodel.Scan;
+import net.sf.mzmine.main.MZmineCore;
+import net.sf.mzmine.parameters.ParameterSet;
+import net.sf.mzmine.parameters.parametertypes.tolerances.MZTolerance;
+import net.sf.mzmine.taskcontrol.AbstractTask;
+import net.sf.mzmine.taskcontrol.TaskStatus;
+import org.openscience.cdk.formula.MolecularFormulaRange;
+
+public class SingleRowIdentificationTask extends AbstractTask {
+  public static final NumberFormat massFormater = MZmineCore.getConfiguration().getMZFormat();
+
+  private int finishedItems = 0, numItems;
+
+  private double searchedMass;
+  private MZTolerance mzTolerance;
+  private int charge;
+  private int numOfResults;
+  private PeakListRow peakListRow;
+  private IonizationType ionType;
+  private MolecularFormulaRange formulaRange;
+  private Double parentMass;
+
+  /**
+   * Create the task.
+   * 
+   * @param parameters task parameters.
+   * @param peakListRow peak-list row to identify.
+   */
+  public SingleRowIdentificationTask(ParameterSet parameters, PeakListRow peakListRow) {
+
+    this.peakListRow = peakListRow;
+
+    searchedMass = parameters.getParameter(NEUTRAL_MASS).getValue();
+    mzTolerance = parameters.getParameter(MZ_TOLERANCE).getValue();
+    numOfResults = parameters.getParameter(MAX_RESULTS).getValue();
+
+    ionType = parameters.getParameter(NEUTRAL_MASS).getIonType();
+    charge = parameters.getParameter(NEUTRAL_MASS).getCharge();
+
+    formulaRange = parameters.getParameter(ELEMENTS).getValue();
+    parentMass = parameters.getParameter(PARENT_MASS).getValue();
+  }
+
+  /**
+   * @see net.sf.mzmine.taskcontrol.Task#getFinishedPercentage()
+   */
+  public double getFinishedPercentage() {
+    if (numItems == 0)
+      return 0;
+    return ((double) finishedItems) / numItems;
+  }
+
+  public String getTaskDescription() {
+    return "Peak identification of " + massFormater.format(searchedMass) + " using Sirius module";
+  }
+
+  /**
+   * @see Runnable#run()
+   */
+  public void run() {
+
+    setStatus(TaskStatus.PROCESSING);
+
+    NumberFormat massFormater = MZmineCore.getConfiguration().getMZFormat();
+
+    ResultWindow window = new ResultWindow(peakListRow, searchedMass, this);
+    window.setTitle("Sirius makes fun " + massFormater.format(searchedMass) + " amu");
+    window.setVisible(true);
+
+    ConstraintsGenerator generator = new ConstraintsGenerator();
+    FormulaConstraints constraints = generator.generateConstraint(formulaRange);
+
+
+    double ppm = mzTolerance.getPpmTolerance();
```

Comment:
This needs updating. MZmine allows specifying both absolute (in m/z) and relative (in ppm) tolerances, and the final tolerance is always maximum of both values. But SIRIUS only works with ppm. So the correct approach is to obtain a mass range by calling mzTolerance.getToleranceRange(searchedMass) and then calculating the ppm value from that range.